[Documentation] Chapter about legacy APIs (#12820)
Sylius/Sylius: docs/book/api/legacy_api.rst

Legacy API
==========

> Whe have decided that we should rebuild out APIs and unify them with API Platform.

Reviewer: ```suggestion
We have decided to rebuild our APIs and unify them with API Platform.
```

> Previously we had 2 separate api for shop (`SyliusShopAPi Plugin) <https://github.com/Sylius/ShopApiPlugin>`_, and for admin (`SyliusAdminApiBundle <https://github.com/Sylius/SyliusAdminApiBundle>`_).

Reviewer: ```suggestion
Previously we had 2 separate APIs for shop (`SyliusShopAPi Plugin) <https://github.com/Sylius/ShopApiPlugin>`_, and for admin (`SyliusAdminApiBundle <https://github.com/Sylius/SyliusAdminApiBundle>`_).
```

Both of them are using the `FOSRestBundle <https://github.com/FriendsOfSymfony/FOSRestBundle>`_, and make operation using commands and events.

> This approach is easy to understand and implement, but when we need to customize something we need to overwrite many files (command, event, commandHandler, EventListener etc).

Reviewer: ```suggestion
This approach is easy to understand and implement, but when we need to customize something we need to overwrite many files (command, event, command handler, event listener etc).
```

> Second reason to create new Sylius API from scratch it fact that the API Platform is a modern framework for API and it replaces FOSRestBundle.

Reviewer: ```suggestion
The second reason to create a new Sylius API from scratch is that the API Platform is a modern framework for API and replaces FOSRestBundle.
```

> We will fix security issues in our legacy APIs but all new feature will be developed only in new API.

Reviewer: ```suggestion
We will fix security issues in our legacy APIs but all new features will be developed only in the new API.
```
